filebeat/input/file_test.go

```diff
@@ -177,6 +180,92 @@ func TestFileEventToMapStrJSON(t *testing.T) {
 				"type": "test_type",
 			},
 		},
+		{
+			// when @timestamp is in JSON and overwrite_keys is true, parse it
```

Comment:
it's often helpful to either print the event or have a test name to be printed in test loop via t.Log in case some test fails.